client/components/button/index.jsx

```diff
@@ -3,49 +3,43 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import assign from 'lodash/assign';
+import { PropTypes, createElement, PureComponent } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
-
-export default React.createClass( {
-
-	displayName: 'Button',
-
-	propTypes: {
-		disabled: React.PropTypes.bool,
-		compact: React.PropTypes.bool,
-		primary: React.PropTypes.bool,
-		scary: React.PropTypes.bool,
-		type: React.PropTypes.string,
-		href: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		borderless: React.PropTypes.bool
-	},
-
-	getDefaultProps() {
-		return {
-			disabled: false,
-			type: 'button',
-			onClick: noop,
-			borderless: false
-		};
-	},
+import { omit } from 'lodash';
+
+export default class Button extends PureComponent {
+	static propTypes = {
+		compact: PropTypes.bool,
+		primary: PropTypes.bool,
+		scary: PropTypes.bool,
+		type: PropTypes.string,
+		href: PropTypes.string,
+		borderless: PropTypes.bool
+	};
+
+	static defaultProps = {
+		type: 'button'
+	};
 
 	render() {
-		const element = this.props.href ? 'a' : 'button';
-		const buttonClasses = classNames( {
-			button: true,
-			'is-compact': this.props.compact,
-			'is-primary': this.props.primary,
-			'is-scary': this.props.scary,
-			'is-borderless': this.props.borderless
+		const omitProps = [ 'compact', 'primary', 'scary', 'borderless' ];
+
+		let tag;
+		if ( this.props.href ) {
+			tag = 'a';
+			omitProps.push( 'type' );
```

Comment:
Because it copies the entire array? If the intention is to mutate the instance, push is a lot faster.